cherrypy/lib/auth_digest.py

```diff
@@ -109,20 +150,23 @@ def get_ha1_file_htdigest(filename):
     argument be an absolute path, to avoid problems.
     """
 
-    def get_ha1(realm, username):
+    def get_ha1(**kwargs):
         result = None
```

Comment:
This way, we can avoid adding even more nesting. It's called a "guard expression".
```suggestion
        if realm is None or username is None:
            return None

        result = None
```